.github/workflows/release-fastlane-android.yml

```diff
@@ -24,8 +24,12 @@ jobs:
       matrix:
         lane: ${{ fromJson(inputs.lanes) }}
     steps:
-      - uses: actions/checkout@v4
-      - uses: google-github-actions/auth@v2
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.branch }}
+      - name: Authenticate to Google Cloud
+        uses: google-github-actions/auth@v2
```

Comment:
checkout must go first so google auth token is available in the subsequent steps.
see docs: https://github.com/google-github-actions/auth#prerequisites

the next checkouts will erase the auth token, so I reduced the steps to a single checkout.